etna/articles/models.py

```diff
@@ -41,6 +41,73 @@
 )
 
 
+@register_snippet
+class ArticleTag(TagBase):
+    free_tagging = False
+    skos_id = models.CharField(
+        blank=True,
+        unique=True,
+        db_index=True,
+        max_length=100,
+        verbose_name="SKOS identifier",
+        help_text="Used as the identifier for this tag when sending page metatdata to the CIIM API.",
+    )
+
+    class Meta:
+        verbose_name = "article tag"
+        verbose_name_plural = "article tags"
+
+    panels = (
+        FieldPanel("name"),
+        FieldPanel("slug"),
+        FieldPanel("skos_id"),
+    )
+
+    def clean(self, *args, **kwargs):
+        if not self.skos_id and self.name:
+            # Generate a unique skos_id value for new tags
+            base_value = skos_id_from_text(self.name)
+            self.skos_id = base_value
+            i = 2
+            while (
+                ArticleTag.objects.exclude(id=self.id)
+                .filter(skos_id=self.skos_id)
+                .exists()
+            ):
+                self.skos_id = f"{base_value[:97]}_{i}"
+                i += 1
+        return super().clean(*args, **kwargs)
+
+
+class TaggedArticle(ItemBase):
+    tag = models.ForeignKey(
+        ArticleTag, related_name="tagged_article", on_delete=models.CASCADE
+    )
+    content_object = ParentalKey(
+        to="wagtailcore.Page",
+        on_delete=models.CASCADE,
+        related_name="tagged_items",
+    )
```

Comment:
Was it necessary to move these methods above? With changes like these it appears to be new ones to review. (only  to="wagtailcore.Page", has changed though)

I think, previously these methods are under ArticleIndexPage as it does which do not use them, and seemed logically placed. 

Will leave it to you if you if you'd like to restore them to their original location.

Reply:
Hi Allan, I think to me it just made sense to have everything at the top but I do get your point 🙂 I think I will leave it there but if you'd like me to move it, then I can - happy for whatever you think🙂

Reply:
I have conveyed my view and As  I mentioned earlier left it to you if you wanted to restore it.